### ISSUE TITLE
Configurable message expiration

### ISSUE THREAD

**Issue by kruthoff** [nextcloud/spreed#8547]
For a long-running chat, I would prefer to set message expiration to 6 Months or to 1 Year, or to define my own retention time, i.e. 180 Days.

It would be cool to expire existing talk messages with this new feature of NC 25. The feature is activated just for new messages, as far as I understand.

Thanks for your support!

**Comment by HuguesDug** [comment]
Hello,

While we would implement more customizing in the "expiring messages", could be nice to give the option to user to get all messages removed either when time limit is reached or ONLY for the ones that have been read by all participants.

So far, I do this while having a event running on the SQL server, changing the date of the expire_date field in table oc_comments. It checks if message id is lower thant the last read message Id for each user in the room, and, if so, I deduct 180 days from the current expire_date. At next nextcloud cron job, the messages are then removed gently. But in everycase, when time is reached, read or not, it goes away, what can be annoying. 

Works fine, except on IOS-talk, that has message caching implemented. So the expire_date is kept to original value on the local system, and read message goes away on these devices only if expiring limit is reached, being read of not. That is annoying from privacy standpoint but also, some users could miss some portions of the conversation.

**Comment by marcotrevisan** [comment]
I would like to add to @kruthoff's requirement: it would be nice to also have the option to delete only the attachments older than "X", leaving the messages (I'm borrowing this from a popular chat server we are using which has this option too and it's very useful).

**Comment by t0maz** [comment]
I miss this feature too.

Additionally it would be great if admin can set default messages expiration time and have ability to enforce max expiration time (users would have ability to set lower one but not higher value).

**Comment by cantrust-hosting-cooperative** [comment]
This is a real blocker to organizations adopting Nextcloud Talk.  4 weeks is not long enough for many operational chats, but endlessly growing database (the microsoft method) is also unnaceptable.  

Is there a technical reason that it's blocked?  Could we not just extend the options to add two more with higher constant values?  Or is there another reason this can't be implemented?

**Comment by cantrust-hosting-cooperative** [comment]
manually setting the message_expiration field on the room's row in oc_talk_rooms allows setting it to whatever value you want.  It then shows in the Room Settings as "Custom expiration".      <code>UPDATE oc_talk SET message_expiration = 31536000 WHERE name LIKE 'My Room Name';</code>

I read on another reddit thread that the main issue with a longer timeout is that devices have to disappear the messages, and setting timeouts longer than 4 weeks on a phone can be problematic (battery life?  unreliable?).   I cannot find this on any of the github threads though.

It would still be great to be able to override this setting if you were not worried about underpowered mobile devices or reliability deleting on the phone.  In my case I'm worried far more about database bloat more than whether the devices reliably delete.  

Ideally there should be choice available to the administrator as Nextclouds come in many shapes and sizes!

**Comment by nickvergessen** [comment]
> I read on another reddit thread that the main issue with a longer timeout is that devices have to disappear the messages, and setting timeouts longer than 4 weeks on a phone can be problematic (battery life? unreliable?). I cannot find this on any of the github threads though.

This is non-sense.

**Comment by damienvancouver** [comment]
> > I read on another reddit thread that the main issue with a longer timeout is that devices have to disappear the messages, and setting timeouts longer than 4 weeks on a phone can be problematic (battery life? unreliable?). I cannot find this on any of the github threads though.
> 
> This is non-sense. Apart from average use-cases there is no reason to not provide a very high value. Maybe we add a custom option in the future, it just does not exist for now.

OK that's good to hear, it did sound a bit silly!   That's what I get for looking at threads on Reddit!    Thanks for tagging this issue, I will maybe take a crack at a patch then.  Thanks also for the clarification!

**Comment by expressrussian** [comment]
Yes, we need auto-delete feature which enable the server to automatically delete the messages in nextcloud talk that are up to 1 year old (4 weeks is not enough). Make this like in Whatsapp/Telegram - configured per chat by any user. Additionally, allow administrators to configure this globally.

**Comment by webkris** [comment]
I would love to be able to set message retention to a custom number / 1 year.
